The job **`x86_64-msvc`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/8041555800/job/21960960191) [(plain)](https://github.com/rust-lang-ci/rust/commit/34d475822edd092651e792ede2fa62e372a11e6d/checks/21960960191/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
  0% 956 - msys64\var\lib\pacman\local\info-7.1-2
                                                 
  0% 1039 - msys64\clangarm64.ico
                                 
  0% 1095 - msys64\usr\bin\addgnupghome
  0% 1182 - msys64\usr\bin\xzfgrep
                                  
  2% 1235 - msys64\usr\lib\perl5\core_perl\CORE\fakesdio.h
                                                          
                                                          
  4% 1289 - msys64\usr\lib\perl5\core_perl\CORE\sbox32_hash.h
                                                             
  4% 1315 - msys64\usr\lib\perl5\core_perl\Digest\SHA.pm
                                                        
  5% 1377 - msys64\usr\lib\perl5\core_perl\List\Util\XS.pm
  5% 1498 - msys64\usr\lib\terminfo\61\aaa-20
                                             
  5% 1521 - msys64\usr\lib\terminfo\61\aaa-60-s
                                               
                                               
  5% 1545 - msys64\usr\lib\terminfo\61\addsviewpoint
                                                    
  5% 1577 - msys64\usr\lib\terminfo\61\alacritty+common
                                                       
  5% 1600 - msys64\usr\lib\terminfo\61\ambassador
  5% 1623 - msys64\usr\lib\terminfo\61\ansi+cup
                                               
  5% 1643 - msys64\usr\lib\terminfo\61\ansi-color-2-emx
                                                       
---
  5% 2289 - msys64\usr\lib\terminfo\64\dwk
                                          
  5% 2365 - msys64\usr\lib\terminfo\67\gnome-rh62
                                                 
  5% 2422 - msys64\usr\lib\terminfo\68\h29a-kc-uc
                                                 
  5% 2423 - msys64\usr\lib\terminfo\68\h29a-nkc-bc
  5% 2574 - msys64\usr\lib\terminfo\69\ibm6153-40
                                                 
  5% 2691 - msys64\usr\lib\terminfo\6c\linux-lat
                                                
---
  5% 3077 - msys64\usr\lib\terminfo\70\pc3-bold
                                               
  5% 3169 - msys64\usr\lib\terminfo\70\pt100
                                            
  6% 3277 - msys64\usr\lib\terminfo\73\scoansi-new
                                                  
  6% 3373 - msys64\usr\lib\terminfo\73\sun-cgsix
  6% 3485 - msys64\usr\lib\terminfo\74\ti916-132
                                                
  6% 3601 - msys64\usr\lib\terminfo\74\tvi920b+fn
                                                 
---
  6% 4093 - msys64\usr\lib\terminfo\78\xnuppc+200x64
                                                    
  6% 4197 - msys64\usr\lib\terminfo\78\xterm-color
                                                  
  6% 4297 - msys64\usr\share\awk\noassign.awk
  6% 4397 - msys64\usr\share\bash-completion\completions\chown
                                                              
  6% 4481 - msys64\usr\share\bash-completion\completions\feh
                                                            
                                                            
  6% 4603 - msys64\usr\share\bash-completion\completions\interdiff
                                                                  
  6% 4687 - msys64\usr\share\bash-completion\completions\lvmdiskscan
                                                                    
  6% 4775 - msys64\usr\share\bash-completion\completions\pdftotext
  7% 4873 - msys64\usr\share\bash-completion\completions\pyvenv-3.6
                                                                   
  7% 5014 - msys64\usr\share\bash-completion\completions\vgextend
                                                                 
                                                                 
  7% 5093 - msys64\usr\share\bash-completion\completions\_udevadm
                                                                 
  8% 5162 - msys64\usr\share\doc\cygwin-ug-net\ov-ex-unix.html
  8% 5219 - msys64\usr\share\doc\libunistring\libunistring_12.html
                                                                  
                                                                  
  8% 5267 - msys64\usr\share\doc\util-linux\getopt\getopt-parse.tcsh
  9% 5352 - msys64\usr\share\doc\xz\README
                                          
                                          
  9% 5430 - msys64\usr\share\icons\hicolor\64x64\apps\mintty.png
 10% 5518 - msys64\usr\share\licenses\libargp\COPYING.LESSER
                                                            
 13% 5643 - msys64\usr\share\locale\da\LC_MESSAGES\gnupg2.mo
                                                            
---
 32% 6499 - msys64\usr\share\makepkg\source\git.sh
                                                  
 32% 6630 - msys64\usr\share\man\man1\chroot.1.gz
                                                 
 32% 6676 - msys64\usr\share\man\man1\dsaparam.1ssl.gz
 32% 6700 - msys64\usr\share\man\man1\factor.1.gz
                                                 
                                                 
 32% 6721 - msys64\usr\share\man\man1\genrsa.1ssl.gz
                                                    
 32% 6730 - msys64\usr\share\man\man1\gmondump.1
 32% 6739 - msys64\usr\share\man\man1\gpgconf.1.gz
                                                  
                                                  
 32% 6845 - msys64\usr\share\man\man1\ocsp.1ssl.gz
                                                  
 33% 6908 - msys64\usr\share\man\man1\pathchk.1.gz
 33% 6990 - msys64\usr\share\man\man1\shred.1.gz
                                                
 33% 7152 - msys64\usr\share\man\man3\asn1_parser2tree.3.gz
                                                           
---
 33% 7367 - msys64\usr\share\man\man3\CURLOPT_FTP_USE_PRET.3.gz
                                                               
 33% 7433 - msys64\usr\share\man\man3\CURLOPT_OPENSOCKETDATA.3.gz
                                                                 
 33% 7525 - msys64\usr\share\man\man3\CURLOPT_SSH_HOSTKEYFUNCTION.3.gz
 33% 7600 - msys64\usr\share\man\man3\CURLSHOPT_UNLOCKFUNC.3.gz
                                                               
 33% 7685 - msys64\usr\share\man\man3\curl_ws_recv.3.gz
                                                       
---
 33% 8088 - msys64\usr\share\man\man3\libcurl-env.3.gz
                                                      
 33% 8212 - msys64\usr\share\man\man3\libssh2_session_init_ex.3.gz
                                                                  
 34% 8331 - msys64\usr\share\man\man3\mouse_trafo.3x.gz
 34% 8443 - msys64\usr\share\man\man3\nl.3x.gz
                                              
 34% 8551 - msys64\usr\share\man\man3\set_field_pad.3x.gz
                                                         
                                                         
 34% 8663 - msys64\usr\share\man\man3\tok_end.3.gz
                                                  
 34% 8783 - msys64\usr\share\man\man3\wgetn_wstr.3x.gz
 34% 8925 - msys64\usr\share\man\pt_BR\man1\unlzma.1.gz
                                                       
                                                       
 38% 9027 - msys64\usr\share\nano\extra\povray.nanorc
 38% 9139 - msys64\usr\share\perl5\core_perl\CPAN\Meta\Spec.pm
                                                              
 39% 9272 - msys64\usr\share\perl5\core_perl\Getopt\Std.pm
                                                          
                                                          
 41% 9405 - msys64\usr\share\perl5\core_perl\Pod\Simple\Progress.pm
                                                                   
 41% 9543 - msys64\usr\share\perl5\core_pe . \Hub\Interceptor\Terminator.pm
 42% 9672 - msys64\usr\share\perl5\core_perl\Unicode\Collate\Locale\zh_pin.pl
                                                                             
                                                                             
 43% 9804 - msys64\usr\share\perl5\core_perl\unicore\lib\GCB\EX.pl
                                                                  
 43% 9903 - msys64\usr\share\perl5\core_perl\unicore\lib\Jg\Lam.pl
                                                                  
 43% 10033 - msys64\usr\share\perl5\core_perl\unicore\lib\Perl\_PerlFol.pl
                                                                          
 43% 10177 - msys64\usr\share\perl5\core_perl\unicore\lib\WB\WSegSpac.pl
 45% 10282 - msys64\usr\share\terminfo\36\605x
                                              
 45% 10425 - msys64\usr\share\terminfo\61\alt5
                                              
---
 45% 10908 - msys64\usr\share\terminfo\64\d414-unix-sr
                                                      
 45% 11035 - msys64\usr\share\terminfo\64\dec-vt340
                                                   
 45% 11155 - msys64\usr\share\terminfo\65\esprit-am
 45% 11297 - msys64\usr\share\terminfo\68\hp236
                                               
 45% 11443 - msys64\usr\share\terminfo\69\ibmvga
                                                
                                                
 45% 11591 - msys64\usr\share\terminfo\6d\mime340
                                                 
 45% 11715 - msys64\usr\share\terminfo\6e\ncsa-vt220-8
 45% 11859 - msys64\usr\share\terminfo\6f\old-st
                                                
 45% 12003 - msys64\usr\share\terminfo\70\pro350
                                                
---
 46% 12395 - msys64\usr\share\terminfo\74\tvi9065
                                                 
 46% 12535 - msys64\usr\share\terminfo\76\vi200-rv
                                                  
 46% 12652 - msys64\usr\share\terminfo\76\vt52+keypad
 46% 12800 - msys64\usr\share\terminfo\77\wy99gt
                                                
 46% 12944 - msys64\usr\share\terminfo\78\xnuppc+c
                                                  
                                                  
 46% 13092 - msys64\usr\share\terminfo\7a\z29b
                                              
 46% 13244 - msys64\usr\share\zoneinfo\America\Indiana\Knox
 46% 13392 - msys64\usr\share\zoneinfo\Asia\Kabul
                                                 
 46% 13529 - msys64\usr\share\zoneinfo\Etc\GMT-3
                                                
                                                
 46% 13645 - msys64\usr\share\zoneinfo\NZ-CHAT
 46% 13785 - msys64\usr\share\zoneinfo\posix\America\Chicago
                                                            
                                                            
 46% 13925 - msys64\usr\share\zoneinfo\posix\Antarctica\Vostok
 46% 14073 - msys64\usr\share\zoneinfo\posix\CET
                                                
                                                
 46% 14221 - msys64\usr\share\zoneinfo\posix\Pacific\Bougainville
 46% 14367 - msys64\usr\share\zoneinfo\right\America\Atka
                                                         
 46% 14504 - msys64\usr\share\zoneinfo\right\America\Thunder_Bay
                                                                
                                                                
 46% 14649 - msys64\usr\share\zoneinfo\right\Australia\Lord_Howe
 46% 14797 - msys64\usr\share\zoneinfo\right\Indian\Comoro
                                                          
 47% 14941 - msys64\var\lib\pacman\local\bash-5.2.021-1\desc
                                                            
---
 50% 15311 - msys64\usr\bin\cygcheck.exe
                                        
 54% 15423 - msys64\usr\bin\lzma.exe
                                    
 63% 15499 - msys64\usr\bin\msys-hdb-9.dll
 80% 15567 - msys64\usr\bin\pacsort.exe
                                       
 92% 15659 - msys64\usr\bin\umount.exe
                                      
                                      
 97% 15779 - msys64\usr\lib\perl5\core_perl\auto\XS\APItest\APItest.dll
Everything is Ok
##[endgroup]
##[group]Disable Key Refresh...
##[endgroup]
---
Testing cranelift stage1 (x86_64-pc-windows-msvc -> x86_64-pc-windows-msvc)
   Compiling y v0.1.0 (C:\a\rust\rust\compiler\rustc_codegen_cranelift\build_system)
[RUSTC-TIMING] y test:false 2.572
    Finished release [optimized] target(s) in 2.68s
     Running `C:\a\rust\rust\build\x86_64-pc-windows-msvc\stage1-codegen\x86_64-pc-windows-msvc\release\y.exe test --download-dir 'C:\a\rust\rust\build\cg_clif_download' --out-dir 'C:\a\rust\rust\build\x86_64-pc-windows-msvc\stage1-tools\cg_clif' --no-unstable-features --use-backend cranelift --sysroot llvm --skip-test testsuite.extended_sysroot`
[BUILD] sysroot None
[RUSTC-TIMING] ____ test:false 0.019
[COPY] stdlib source
[PATCH] "stdlib" <- "0027-stdlib-128bit-atomic-operations.patch"
---
error: make failed
status: exit code: 2
command: "make"
--- stdout -------------------------------
make[1]: Entering directory '/c/a/rust/rust/tests/run-make/arguments-non-c-like-enum'
PATH="/c/a/rust/rust/build/x86_64-pc-windows-msvc/test/run-make/arguments-non-c-like-enum/arguments-non-c-like-enum:C:\a\rust\rust\build\x86_64-pc-windows-msvc\stage2\bin:/c/Program Files (x86)/Windows Kits/10/bin/x64:/c/Program Files (x86)/Windows Kits/10/bin/10.0.22621.0/x64:/c/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.29.30133/bin/HostX64/x64:/c/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.29.30133/bin/HostX64/x64:/c/a/rust/rust/build/x86_64-pc-windows-msvc/stage0-bootstrap-tools/x86_64-pc-windows-msvc/release/deps:/c/a/rust/rust/build/x86_64-pc-windows-msvc/stage0/bin:/c/Program Files/PowerShell/7:/mingw64/bin:/usr/local/bin:/usr/bin:/usr/bin:/c/a/rust/rust/ninja:/c/hostedtoolcache/windows/Python/3.12.2/x64/Scripts:/c/hostedtoolcache/windows/Python/3.12.2/x64:/c/a/rust/rust/sccache:/c/a/_temp/setup-msys2:/c/Program Files/MongoDB/Server/5.0/bin:/c/aliyun-cli:/c/vcpkg:/c/cf-cli:/c/Program Files (x86)/NSIS:/c/tools/zstd:/c/Program Files/Mercurial:/c/hostedtoolcache/windows/stack/2.15.1/x64:/c/cabal/bin:/c/ghcup/bin:/c/mingw64/bin:/c/Program Files/dotnet:/c/Program Files/MySQL/MySQL Server 5.7/bin:/c/Program Files/R/R-4.3.2/bin/x64:/c/SeleniumWebDrivers/GeckoDriver:/c/SeleniumWebDrivers/EdgeDriver:/c/SeleniumWebDrivers/ChromeDriver:/c/Program Files (x86)/sbt/bin:/c/Program Files (x86)/GitHub CLI:/c/Program Files/Git/bin:/c/Program Files (x86)/pipx_bin:/c/npm/prefix:/c/hostedtoolcache/windows/go/1.20.14/x64/bin:/c/hostedtoolcache/windows/Python/3.7.9/x64/Scripts:/c/hostedtoolcache/windows/Python/3.7.9/x64:/c/hostedtoolcache/windows/Ruby/2.5.9/x64/bin:/c/Program Files/OpenSSL/bin:/c/tools/kotlinc/bin:/c/hostedtoolcache/windows/Java_Temurin-Hotspot_jdk/8.0.402-6/x64/bin:/c/Program Files/ImageMagick-7.1.1-Q16-HDRI:/c/Program Files/Microsoft SDKs/Azure/CLI2/wbin:/c/ProgramData/kind:/c/Program Files/Eclipse Foundation/jdk-8.0.302.8-hotspot/bin:/c/ProgramData/Chocolatey/bin:/c/Windows/system32:/c/Windows:/c/Windows/System32/Wbem:/c/Windows/System32/WindowsPowerShell/v1.0:/c/Windows/System32/OpenSSH:/c/Program Files/PowerShell/7:/c/Program Files/Microsoft/Web Platform Installer:/c/Program Files/Microsoft SQL Server/130/Tools/Binn:/c/Program Files/Microsoft SQL Server/Client SDK/ODBC/170/Tools/Binn:/c/Program Files (x86)/Windows Kits/10/Windows Performance Toolkit:/c/Program Files (x86)/Microsoft SQL Server/110/DTS/Binn:/c/Program Files (x86)/Microsoft SQL Server/120/DTS/Binn:/c/Program Files (x86)/Microsoft SQL Server/130/DTS/Binn:/c/Program Files (x86)/Microsoft SQL Server/140/DTS/Binn:/c/Program Files (x86)/Microsoft SQL Server/150/DTS/Binn:/c/Program Files (x86)/Microsoft SQL Server/160/DTS/Binn:/c/Strawberry/c/bin:/c/Strawberry/perl/site/bin:/c/Strawberry/perl/bin:/c/ProgramData/chocolatey/lib/pulumi/tools/Pulumi/bin:/c/Program Files/TortoiseSVN/bin:/c/Program Files/CMake/bin:/c/ProgramData/chocolatey/lib/maven/apache-maven-3.8.7/bin:/c/Program Files/Microsoft Service Fabric/bin/Fabric/Fabric.Code:/c/Program Files/Microsoft SDKs/Service Fabric/Tools/ServiceFabricLocalClusterManager:/c/Program Files/nodejs:/c/Program Files/Git/cmd:/c/Program Files/Git/mingw64/bin:/c/Program Files/Git/usr/bin:/c/Program Files/GitHub CLI:/c/tools/php:/c/Program Files (x86)/sbt/bin:/c/Program Files/Amazon/AWSCLIV2:/c/Program Files/Amazon/SessionManagerPlugin/bin:/c/Program Files/Amazon/AWSSAMCLI/bin:/c/Program Files (x86)/Google/Cloud SDK/google-cloud-sdk/bin:/c/Program Files (x86)/Microsoft BizTalk Server:/c/Program Files/LLVM/bin:/c/Users/runneradmin/.dotnet/tools:/c/Users/runneradmin/.cargo/bin:/c/Users/runneradmin/AppData/Local/Microsoft/WindowsApps:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl" 'C:\a\rust\rust\build\x86_64-pc-windows-msvc\stage2\bin\rustc.exe' --out-dir /c/a/rust/rust/build/x86_64-pc-windows-msvc/test/run-make/arguments-non-c-like-enum/arguments-non-c-like-enum -L /c/a/rust/rust/build/x86_64-pc-windows-msvc/test/run-make/arguments-non-c-like-enum/arguments-non-c-like-enum  -Ainternal_features --crate-type=staticlib nonclike.rs
'C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.29.30133\bin\HostX64\x64\cl.exe' -nologo -MT -Brepro test.c /c/a/rust/rust/build/x86_64-pc-windows-msvc/test/run-make/arguments-non-c-like-enum/arguments-non-c-like-enum/nonclike.lib -Fe:`cygpath -w /c/a/rust/rust/build/x86_64-pc-windows-msvc/test/run-make/arguments-non-c-like-enum/arguments-non-c-like-enum/test.exe` -Fo:`cygpath -w /c/a/rust/rust/build/x86_64-pc-windows-msvc/test/run-make/arguments-non-c-like-enum/arguments-non-c-like-enum/test.obj` \
 ws2_32.lib userenv.lib advapi32.lib bcrypt.lib ntdll.lib 
test.c
nonclike.lib(std-d87c926c681a286a.std.565288a8484ebc3a-cgu.0.rcgu.o) : error LNK2019: unresolved external symbol __imp_WakeByAddressSingle referenced in function _ZN4core3ptr56drop_in_place$LT$std..thread..Packet$LT$$LP$$RP$$GT$$GT$17he58d73a7e37c17dbE
nonclike.lib(std-d87c926c681a286a.std.565288a8484ebc3a-cgu.0.rcgu.o) : error LNK2019: unresolved external symbol __imp_WaitOnAddress referenced in function _ZN3std6thread4park17h004306a0b01b0c4cE
C:\a\rust\rust\build\x86_64-pc-windows-msvc\test\run-make\arguments-non-c-like-enum\arguments-non-c-like-enum\test.exe : fatal error LNK1120: 2 unresolved externals
make[1]: Leaving directory '/c/a/rust/rust/tests/run-make/arguments-non-c-like-enum'
--- stderr -------------------------------
make[1]: *** [Makefile:6: all] Error 2
------------------------------------------



---- [run-make] tests\run-make\c-link-to-rust-va-list-fn stdout ----

error: make failed
status: exit code: 2
command: "make"
--- stdout -------------------------------
make[1]: Entering directory '/c/a/rust/rust/tests/run-make/c-link-to-rust-va-list-fn'
PATH="/c/a/rust/rust/build/x86_64-pc-windows-msvc/test/run-make/c-link-to-rust-va-list-fn/c-link-to-rust-va-list-fn:C:\a\rust\rust\build\x86_64-pc-windows-msvc\stage2\bin:/c/Program Files (x86)/Windows Kits/10/bin/x64:/c/Program Files (x86)/Windows Kits/10/bin/10.0.22621.0/x64:/c/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.29.30133/bin/HostX64/x64:/c/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.29.30133/bin/HostX64/x64:/c/a/rust/rust/build/x86_64-pc-windows-msvc/stage0-bootstrap-tools/x86_64-pc-windows-msvc/release/deps:/c/a/rust/rust/build/x86_64-pc-windows-msvc/stage0/bin:/c/Program Files/PowerShell/7:/mingw64/bin:/usr/local/bin:/usr/bin:/usr/bin:/c/a/rust/rust/ninja:/c/hostedtoolcache/windows/Python/3.12.2/x64/Scripts:/c/hostedtoolcache/windows/Python/3.12.2/x64:/c/a/rust/rust/sccache:/c/a/_temp/setup-msys2:/c/Program Files/MongoDB/Server/5.0/bin:/c/aliyun-cli:/c/vcpkg:/c/cf-cli:/c/Program Files (x86)/NSIS:/c/tools/zstd:/c/Program Files/Mercurial:/c/hostedtoolcache/windows/stack/2.15.1/x64:/c/cabal/bin:/c/ghcup/bin:/c/mingw64/bin:/c/Program Files/dotnet:/c/Program Files/MySQL/MySQL Server 5.7/bin:/c/Program Files/R/R-4.3.2/bin/x64:/c/SeleniumWebDrivers/GeckoDriver:/c/SeleniumWebDrivers/EdgeDriver:/c/SeleniumWebDrivers/ChromeDriver:/c/Program Files (x86)/sbt/bin:/c/Program Files (x86)/GitHub CLI:/c/Program Files/Git/bin:/c/Program Files (x86)/pipx_bin:/c/npm/prefix:/c/hostedtoolcache/windows/go/1.20.14/x64/bin:/c/hostedtoolcache/windows/Python/3.7.9/x64/Scripts:/c/hostedtoolcache/windows/Python/3.7.9/x64:/c/hostedtoolcache/windows/Ruby/2.5.9/x64/bin:/c/Program Files/OpenSSL/bin:/c/tools/kotlinc/bin:/c/hostedtoolcache/windows/Java_Temurin-Hotspot_jdk/8.0.402-6/x64/bin:/c/Program Files/ImageMagick-7.1.1-Q16-HDRI:/c/Program Files/Microsoft SDKs/Azure/CLI2/wbin:/c/ProgramData/kind:/c/Program Files/Eclipse Foundation/jdk-8.0.302.8-hotspot/bin:/c/ProgramData/Chocolatey/bin:/c/Windows/system32:/c/Windows:/c/Windows/System32/Wbem:/c/Windows/System32/WindowsPowerShell/v1.0:/c/Windows/System32/OpenSSH:/c/Program Files/PowerShell/7:/c/Program Files/Microsoft/Web Platform Installer:/c/Program Files/Microsoft SQL Server/130/Tools/Binn:/c/Program Files/Microsoft SQL Server/Client SDK/ODBC/170/Tools/Binn:/c/Program Files (x86)/Windows Kits/10/Windows Performance Toolkit:/c/Program Files (x86)/Microsoft SQL Server/110/DTS/Binn:/c/Program Files (x86)/Microsoft SQL Server/120/DTS/Binn:/c/Program Files (x86)/Microsoft SQL Server/130/DTS/Binn:/c/Program Files (x86)/Microsoft SQL Server/140/DTS/Binn:/c/Program Files (x86)/Microsoft SQL Server/150/DTS/Binn:/c/Program Files (x86)/Microsoft SQL Server/160/DTS/Binn:/c/Strawberry/c/bin:/c/Strawberry/perl/site/bin:/c/Strawberry/perl/bin:/c/ProgramData/chocolatey/lib/pulumi/tools/Pulumi/bin:/c/Program Files/TortoiseSVN/bin:/c/Program Files/CMake/bin:/c/ProgramData/chocolatey/lib/maven/apache-maven-3.8.7/bin:/c/Program Files/Microsoft Service Fabric/bin/Fabric/Fabric.Code:/c/Program Files/Microsoft SDKs/Service Fabric/Tools/ServiceFabricLocalClusterManager:/c/Program Files/nodejs:/c/Program Files/Git/cmd:/c/Program Files/Git/mingw64/bin:/c/Program Files/Git/usr/bin:/c/Program Files/GitHub CLI:/c/tools/php:/c/Program Files (x86)/sbt/bin:/c/Program Files/Amazon/AWSCLIV2:/c/Program Files/Amazon/SessionManagerPlugin/bin:/c/Program Files/Amazon/AWSSAMCLI/bin:/c/Program Files (x86)/Google/Cloud SDK/google-cloud-sdk/bin:/c/Program Files (x86)/Microsoft BizTalk Server:/c/Program Files/LLVM/bin:/c/Users/runneradmin/.dotnet/tools:/c/Users/runneradmin/.cargo/bin:/c/Users/runneradmin/AppData/Local/Microsoft/WindowsApps:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl" 'C:\a\rust\rust\build\x86_64-pc-windows-msvc\stage2\bin\rustc.exe' --out-dir /c/a/rust/rust/build/x86_64-pc-windows-msvc/test/run-make/c-link-to-rust-va-list-fn/c-link-to-rust-va-list-fn -L /c/a/rust/rust/build/x86_64-pc-windows-msvc/test/run-make/c-link-to-rust-va-list-fn/c-link-to-rust-va-list-fn  -Ainternal_features checkrust.rs
'C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.29.30133\bin\HostX64\x64\cl.exe' -nologo -MT -Brepro test.c /c/a/rust/rust/build/x86_64-pc-windows-msvc/test/run-make/c-link-to-rust-va-list-fn/c-link-to-rust-va-list-fn/checkrust.lib -Fe:`cygpath -w /c/a/rust/rust/build/x86_64-pc-windows-msvc/test/run-make/c-link-to-rust-va-list-fn/c-link-to-rust-va-list-fn/test.exe` -Fo:`cygpath -w /c/a/rust/rust/build/x86_64-pc-windows-msvc/test/run-make/c-link-to-rust-va-list-fn/c-link-to-rust-va-list-fn/test.obj` ws2_32.lib userenv.lib advapi32.lib bcrypt.lib ntdll.lib
test.c
checkrust.lib(std-d87c926c681a286a.std.565288a8484ebc3a-cgu.0.rcgu.o) : error LNK2019: unresolved external symbol __imp_WakeByAddressSingle referenced in function _ZN4core3ptr56drop_in_place$LT$std..thread..Packet$LT$$LP$$RP$$GT$$GT$17he58d73a7e37c17dbE
checkrust.lib(std-d87c926c681a286a.std.565288a8484ebc3a-cgu.0.rcgu.o) : error LNK2019: unresolved external symbol __imp_WaitOnAddress referenced in function _ZN3std6thread4park17h004306a0b01b0c4cE
C:\a\rust\rust\build\x86_64-pc-windows-msvc\test\run-make\c-link-to-rust-va-list-fn\c-link-to-rust-va-list-fn\test.exe : fatal error LNK1120: 2 unresolved externals
make[1]: Leaving directory '/c/a/rust/rust/tests/run-make/c-link-to-rust-va-list-fn'
------------------------------------------
--- stderr -------------------------------
make[1]: *** [Makefile:6: all] Error 2
---
error: make failed
status: exit code: 2
command: "make"
--- stdout -------------------------------
make[1]: Entering directory '/c/a/rust/rust/tests/run-make/issue-14500'
PATH="/c/a/rust/rust/build/x86_64-pc-windows-msvc/test/run-make/issue-14500/issue-14500:C:\a\rust\rust\build\x86_64-pc-windows-msvc\stage2\bin:/c/Program Files (x86)/Windows Kits/10/bin/x64:/c/Program Files (x86)/Windows Kits/10/bin/10.0.22621.0/x64:/c/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.29.30133/bin/HostX64/x64:/c/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.29.30133/bin/HostX64/x64:/c/a/rust/rust/build/x86_64-pc-windows-msvc/stage0-bootstrap-tools/x86_64-pc-windows-msvc/release/deps:/c/a/rust/rust/build/x86_64-pc-windows-msvc/stage0/bin:/c/Program Files/PowerShell/7:/mingw64/bin:/usr/local/bin:/usr/bin:/usr/bin:/c/a/rust/rust/ninja:/c/hostedtoolcache/windows/Python/3.12.2/x64/Scripts:/c/hostedtoolcache/windows/Python/3.12.2/x64:/c/a/rust/rust/sccache:/c/a/_temp/setup-msys2:/c/Program Files/MongoDB/Server/5.0/bin:/c/aliyun-cli:/c/vcpkg:/c/cf-cli:/c/Program Files (x86)/NSIS:/c/tools/zstd:/c/Program Files/Mercurial:/c/hostedtoolcache/windows/stack/2.15.1/x64:/c/cabal/bin:/c/ghcup/bin:/c/mingw64/bin:/c/Program Files/dotnet:/c/Program Files/MySQL/MySQL Server 5.7/bin:/c/Program Files/R/R-4.3.2/bin/x64:/c/SeleniumWebDrivers/GeckoDriver:/c/SeleniumWebDrivers/EdgeDriver:/c/SeleniumWebDrivers/ChromeDriver:/c/Program Files (x86)/sbt/bin:/c/Program Files (x86)/GitHub CLI:/c/Program Files/Git/bin:/c/Program Files (x86)/pipx_bin:/c/npm/prefix:/c/hostedtoolcache/windows/go/1.20.14/x64/bin:/c/hostedtoolcache/windows/Python/3.7.9/x64/Scripts:/c/hostedtoolcache/windows/Python/3.7.9/x64:/c/hostedtoolcache/windows/Ruby/2.5.9/x64/bin:/c/Program Files/OpenSSL/bin:/c/tools/kotlinc/bin:/c/hostedtoolcache/windows/Java_Temurin-Hotspot_jdk/8.0.402-6/x64/bin:/c/Program Files/ImageMagick-7.1.1-Q16-HDRI:/c/Program Files/Microsoft SDKs/Azure/CLI2/wbin:/c/ProgramData/kind:/c/Program Files/Eclipse Foundation/jdk-8.0.302.8-hotspot/bin:/c/ProgramData/Chocolatey/bin:/c/Windows/system32:/c/Windows:/c/Windows/System32/Wbem:/c/Windows/System32/WindowsPowerShell/v1.0:/c/Windows/System32/OpenSSH:/c/Program Files/PowerShell/7:/c/Program Files/Microsoft/Web Platform Installer:/c/Program Files/Microsoft SQL Server/130/Tools/Binn:/c/Program Files/Microsoft SQL Server/Client SDK/ODBC/170/Tools/Binn:/c/Program Files (x86)/Windows Kits/10/Windows Performance Toolkit:/c/Program Files (x86)/Microsoft SQL Server/110/DTS/Binn:/c/Program Files (x86)/Microsoft SQL Server/120/DTS/Binn:/c/Program Files (x86)/Microsoft SQL Server/130/DTS/Binn:/c/Program Files (x86)/Microsoft SQL Server/140/DTS/Binn:/c/Program Files (x86)/Microsoft SQL Server/150/DTS/Binn:/c/Program Files (x86)/Microsoft SQL Server/160/DTS/Binn:/c/Strawberry/c/bin:/c/Strawberry/perl/site/bin:/c/Strawberry/perl/bin:/c/ProgramData/chocolatey/lib/pulumi/tools/Pulumi/bin:/c/Program Files/TortoiseSVN/bin:/c/Program Files/CMake/bin:/c/ProgramData/chocolatey/lib/maven/apache-maven-3.8.7/bin:/c/Program Files/Microsoft Service Fabric/bin/Fabric/Fabric.Code:/c/Program Files/Microsoft SDKs/Service Fabric/Tools/ServiceFabricLocalClusterManager:/c/Program Files/nodejs:/c/Program Files/Git/cmd:/c/Program Files/Git/mingw64/bin:/c/Program Files/Git/usr/bin:/c/Program Files/GitHub CLI:/c/tools/php:/c/Program Files (x86)/sbt/bin:/c/Program Files/Amazon/AWSCLIV2:/c/Program Files/Amazon/SessionManagerPlugin/bin:/c/Program Files/Amazon/AWSSAMCLI/bin:/c/Program Files (x86)/Google/Cloud SDK/google-cloud-sdk/bin:/c/Program Files (x86)/Microsoft BizTalk Server:/c/Program Files/LLVM/bin:/c/Users/runneradmin/.dotnet/tools:/c/Users/runneradmin/.cargo/bin:/c/Users/runneradmin/AppData/Local/Microsoft/WindowsApps:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl" 'C:\a\rust\rust\build\x86_64-pc-windows-msvc\stage2\bin\rustc.exe' --out-dir /c/a/rust/rust/build/x86_64-pc-windows-msvc/test/run-make/issue-14500/issue-14500 -L /c/a/rust/rust/build/x86_64-pc-windows-msvc/test/run-make/issue-14500/issue-14500  -Ainternal_features foo.rs --crate-type=rlib
PATH="/c/a/rust/rust/build/x86_64-pc-windows-msvc/test/run-make/issue-14500/issue-14500:C:\a\rust\rust\build\x86_64-pc-windows-msvc\stage2\bin:/c/Program Files (x86)/Windows Kits/10/bin/x64:/c/Program Files (x86)/Windows Kits/10/bin/10.0.22621.0/x64:/c/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.29.30133/bin/HostX64/x64:/c/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.29.30133/bin/HostX64/x64:/c/a/rust/rust/build/x86_64-pc-windows-msvc/stage0-bootstrap-tools/x86_64-pc-windows-msvc/release/deps:/c/a/rust/rust/build/x86_64-pc-windows-msvc/stage0/bin:/c/Program Files/PowerShell/7:/mingw64/bin:/usr/local/bin:/usr/bin:/usr/bin:/c/a/rust/rust/ninja:/c/hostedtoolcache/windows/Python/3.12.2/x64/Scripts:/c/hostedtoolcache/windows/Python/3.12.2/x64:/c/a/rust/rust/sccache:/c/a/_temp/setup-msys2:/c/Program Files/MongoDB/Server/5.0/bin:/c/aliyun-cli:/c/vcpkg:/c/cf-cli:/c/Program Files (x86)/NSIS:/c/tools/zstd:/c/Program Files/Mercurial:/c/hostedtoolcache/windows/stack/2.15.1/x64:/c/cabal/bin:/c/ghcup/bin:/c/mingw64/bin:/c/Program Files/dotnet:/c/Program Files/MySQL/MySQL Server 5.7/bin:/c/Program Files/R/R-4.3.2/bin/x64:/c/SeleniumWebDrivers/GeckoDriver:/c/SeleniumWebDrivers/EdgeDriver:/c/SeleniumWebDrivers/ChromeDriver:/c/Program Files (x86)/sbt/bin:/c/Program Files (x86)/GitHub CLI:/c/Program Files/Git/bin:/c/Program Files (x86)/pipx_bin:/c/npm/prefix:/c/hostedtoolcache/windows/go/1.20.14/x64/bin:/c/hostedtoolcache/windows/Python/3.7.9/x64/Scripts:/c/hostedtoolcache/windows/Python/3.7.9/x64:/c/hostedtoolcache/windows/Ruby/2.5.9/x64/bin:/c/Program Files/OpenSSL/bin:/c/tools/kotlinc/bin:/c/hostedtoolcache/windows/Java_Temurin-Hotspot_jdk/8.0.402-6/x64/bin:/c/Program Files/ImageMagick-7.1.1-Q16-HDRI:/c/Program Files/Microsoft SDKs/Azure/CLI2/wbin:/c/ProgramData/kind:/c/Program Files/Eclipse Foundation/jdk-8.0.302.8-hotspot/bin:/c/ProgramData/Chocolatey/bin:/c/Windows/system32:/c/Windows:/c/Windows/System32/Wbem:/c/Windows/System32/WindowsPowerShell/v1.0:/c/Windows/System32/OpenSSH:/c/Program Files/PowerShell/7:/c/Program Files/Microsoft/Web Platform Installer:/c/Program Files/Microsoft SQL Server/130/Tools/Binn:/c/Program Files/Microsoft SQL Server/Client SDK/ODBC/170/Tools/Binn:/c/Program Files (x86)/Windows Kits/10/Windows Performance Toolkit:/c/Program Files (x86)/Microsoft SQL Server/110/DTS/Binn:/c/Program Files (x86)/Microsoft SQL Server/120/DTS/Binn:/c/Program Files (x86)/Microsoft SQL Server/130/DTS/Binn:/c/Program Files (x86)/Microsoft SQL Server/140/DTS/Binn:/c/Program Files (x86)/Microsoft SQL Server/150/DTS/Binn:/c/Program Files (x86)/Microsoft SQL Server/160/DTS/Binn:/c/Strawberry/c/bin:/c/Strawberry/perl/site/bin:/c/Strawberry/perl/bin:/c/ProgramData/chocolatey/lib/pulumi/tools/Pulumi/bin:/c/Program Files/TortoiseSVN/bin:/c/Program Files/CMake/bin:/c/ProgramData/chocolatey/lib/maven/apache-maven-3.8.7/bin:/c/Program Files/Microsoft Service Fabric/bin/Fabric/Fabric.Code:/c/Program Files/Microsoft SDKs/Service Fabric/Tools/ServiceFabricLocalClusterManager:/c/Program Files/nodejs:/c/Program Files/Git/cmd:/c/Program Files/Git/mingw64/bin:/c/Program Files/Git/usr/bin:/c/Program Files/GitHub CLI:/c/tools/php:/c/Program Files (x86)/sbt/bin:/c/Program Files/Amazon/AWSCLIV2:/c/Program Files/Amazon/SessionManagerPlugin/bin:/c/Program Files/Amazon/AWSSAMCLI/bin:/c/Program Files (x86)/Google/Cloud SDK/google-cloud-sdk/bin:/c/Program Files (x86)/Microsoft BizTalk Server:/c/Program Files/LLVM/bin:/c/Users/runneradmin/.dotnet/tools:/c/Users/runneradmin/.cargo/bin:/c/Users/runneradmin/AppData/Local/Microsoft/WindowsApps:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl" 'C:\a\rust\rust\build\x86_64-pc-windows-msvc\stage2\bin\rustc.exe' --out-dir /c/a/rust/rust/build/x86_64-pc-windows-msvc/test/run-make/issue-14500/issue-14500 -L /c/a/rust/rust/build/x86_64-pc-windows-msvc/test/run-make/issue-14500/issue-14500  -Ainternal_features bar.rs --crate-type=staticlib -C lto -L. -o /c/a/rust/rust/build/x86_64-pc-windows-msvc/test/run-make/issue-14500/issue-14500/libbar.a
'C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.29.30133\bin\HostX64\x64\cl.exe' -nologo -MT -Brepro foo.c /c/a/rust/rust/build/x86_64-pc-windows-msvc/test/run-make/issue-14500/issue-14500/libbar.a ws2_32.lib userenv.lib advapi32.lib bcrypt.lib ntdll.lib -Fe:`cygpath -w /c/a/rust/rust/build/x86_64-pc-windows-msvc/test/run-make/issue-14500/issue-14500/foo.exe` -Fo:`cygpath -w /c/a/rust/rust/build/x86_64-pc-windows-msvc/test/run-make/issue-14500/issue-14500/foo.obj`
foo.c
libbar.a(libbar.bar.67338c9292c4797c-cgu.0.rcgu.o) : error LNK2019: unresolved external symbol __imp_WaitOnAddress referenced in function _ZN3std10sys_common4once5queue4Once4call17h4d15eaa35c1bfcb7E
libbar.a(libbar.bar.67338c9292c4797c-cgu.0.rcgu.o) : error LNK2019: unresolved external symbol __imp_WakeByAddressSingle referenced in function _ZN83_$LT$std..sys_common..once..queue..WaiterQueue$u20$as$u20$core..ops..drop..Drop$GT$4drop17h3b920a1f9cb93053E
C:\a\rust\rust\build\x86_64-pc-windows-msvc\test\run-make\issue-14500\issue-14500\foo.exe : fatal error LNK1120: 2 unresolved externals
make[1]: Leaving directory '/c/a/rust/rust/tests/run-make/issue-14500'
--- stderr -------------------------------
warning: ignoring --out-dir flag due to -o flag

warning: 1 warning emitted
warning: 1 warning emitted

cl : Command line warning D9024 : unrecognized source file type 'C:/a/rust/rust/build/x86_64-pc-windows-msvc/test/run-make/issue-14500/issue-14500/libbar.a', object file assumed
make[1]: *** [Makefile:14: all] Error 2


---- [run-make] tests\run-make\lto-smoke-c stdout ----


error: make failed
status: exit code: 2
command: "make"
--- stdout -------------------------------
make[1]: Entering directory '/c/a/rust/rust/tests/run-make/lto-smoke-c'
PATH="/c/a/rust/rust/build/x86_64-pc-windows-msvc/test/run-make/lto-smoke-c/lto-smoke-c:C:\a\rust\rust\build\x86_64-pc-windows-msvc\stage2\bin:/c/Program Files (x86)/Windows Kits/10/bin/x64:/c/Program Files (x86)/Windows Kits/10/bin/10.0.22621.0/x64:/c/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.29.30133/bin/HostX64/x64:/c/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.29.30133/bin/HostX64/x64:/c/a/rust/rust/build/x86_64-pc-windows-msvc/stage0-bootstrap-tools/x86_64-pc-windows-msvc/release/deps:/c/a/rust/rust/build/x86_64-pc-windows-msvc/stage0/bin:/c/Program Files/PowerShell/7:/mingw64/bin:/usr/local/bin:/usr/bin:/usr/bin:/c/a/rust/rust/ninja:/c/hostedtoolcache/windows/Python/3.12.2/x64/Scripts:/c/hostedtoolcache/windows/Python/3.12.2/x64:/c/a/rust/rust/sccache:/c/a/_temp/setup-msys2:/c/Program Files/MongoDB/Server/5.0/bin:/c/aliyun-cli:/c/vcpkg:/c/cf-cli:/c/Program Files (x86)/NSIS:/c/tools/zstd:/c/Program Files/Mercurial:/c/hostedtoolcache/windows/stack/2.15.1/x64:/c/cabal/bin:/c/ghcup/bin:/c/mingw64/bin:/c/Program Files/dotnet:/c/Program Files/MySQL/MySQL Server 5.7/bin:/c/Program Files/R/R-4.3.2/bin/x64:/c/SeleniumWebDrivers/GeckoDriver:/c/SeleniumWebDrivers/EdgeDriver:/c/SeleniumWebDrivers/ChromeDriver:/c/Program Files (x86)/sbt/bin:/c/Program Files (x86)/GitHub CLI:/c/Program Files/Git/bin:/c/Program Files (x86)/pipx_bin:/c/npm/prefix:/c/hostedtoolcache/windows/go/1.20.14/x64/bin:/c/hostedtoolcache/windows/Python/3.7.9/x64/Scripts:/c/hostedtoolcache/windows/Python/3.7.9/x64:/c/hostedtoolcache/windows/Ruby/2.5.9/x64/bin:/c/Program Files/OpenSSL/bin:/c/tools/kotlinc/bin:/c/hostedtoolcache/windows/Java_Temurin-Hotspot_jdk/8.0.402-6/x64/bin:/c/Program Files/ImageMagick-7.1.1-Q16-HDRI:/c/Program Files/Microsoft SDKs/Azure/CLI2/wbin:/c/ProgramData/kind:/c/Program Files/Eclipse Foundation/jdk-8.0.302.8-hotspot/bin:/c/ProgramData/Chocolatey/bin:/c/Windows/system32:/c/Windows:/c/Windows/System32/Wbem:/c/Windows/System32/WindowsPowerShell/v1.0:/c/Windows/System32/OpenSSH:/c/Program Files/PowerShell/7:/c/Program Files/Microsoft/Web Platform Installer:/c/Program Files/Microsoft SQL Server/130/Tools/Binn:/c/Program Files/Microsoft SQL Server/Client SDK/ODBC/170/Tools/Binn:/c/Program Files (x86)/Windows Kits/10/Windows Performance Toolkit:/c/Program Files (x86)/Microsoft SQL Server/110/DTS/Binn:/c/Program Files (x86)/Microsoft SQL Server/120/DTS/Binn:/c/Program Files (x86)/Microsoft SQL Server/130/DTS/Binn:/c/Program Files (x86)/Microsoft SQL Server/140/DTS/Binn:/c/Program Files (x86)/Microsoft SQL Server/150/DTS/Binn:/c/Program Files (x86)/Microsoft SQL Server/160/DTS/Binn:/c/Strawberry/c/bin:/c/Strawberry/perl/site/bin:/c/Strawberry/perl/bin:/c/ProgramData/chocolatey/lib/pulumi/tools/Pulumi/bin:/c/Program Files/TortoiseSVN/bin:/c/Program Files/CMake/bin:/c/ProgramData/chocolatey/lib/maven/apache-maven-3.8.7/bin:/c/Program Files/Microsoft Service Fabric/bin/Fabric/Fabric.Code:/c/Program Files/Microsoft SDKs/Service Fabric/Tools/ServiceFabricLocalClusterManager:/c/Program Files/nodejs:/c/Program Files/Git/cmd:/c/Program Files/Git/mingw64/bin:/c/Program Files/Git/usr/bin:/c/Program Files/GitHub CLI:/c/tools/php:/c/Program Files (x86)/sbt/bin:/c/Program Files/Amazon/AWSCLIV2:/c/Program Files/Amazon/SessionManagerPlugin/bin:/c/Program Files/Amazon/AWSSAMCLI/bin:/c/Program Files (x86)/Google/Cloud SDK/google-cloud-sdk/bin:/c/Program Files (x86)/Microsoft BizTalk Server:/c/Program Files/LLVM/bin:/c/Users/runneradmin/.dotnet/tools:/c/Users/runneradmin/.cargo/bin:/c/Users/runneradmin/AppData/Local/Microsoft/WindowsApps:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl" 'C:\a\rust\rust\build\x86_64-pc-windows-msvc\stage2\bin\rustc.exe' --out-dir /c/a/rust/rust/build/x86_64-pc-windows-msvc/test/run-make/lto-smoke-c/lto-smoke-c -L /c/a/rust/rust/build/x86_64-pc-windows-msvc/test/run-make/lto-smoke-c/lto-smoke-c  -Ainternal_features foo.rs -C lto
'C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.29.30133\bin\HostX64\x64\cl.exe' -nologo -MT -Brepro bar.c /c/a/rust/rust/build/x86_64-pc-windows-msvc/test/run-make/lto-smoke-c/lto-smoke-c/foo.lib \
 -Fe:`cygpath -w /c/a/rust/rust/build/x86_64-pc-windows-msvc/test/run-make/lto-smoke-c/lto-smoke-c/bar.exe` -Fo:`cygpath -w /c/a/rust/rust/build/x86_64-pc-windows-msvc/test/run-make/lto-smoke-c/lto-smoke-c/bar.obj` \
 ws2_32.lib userenv.lib advapi32.lib bcrypt.lib ntdll.lib 
bar.c
foo.lib(foo.foo.5be5606e1f6aa79b-cgu.0.rcgu.o) : error LNK2019: unresolved external symbol __imp_WaitOnAddress referenced in function _ZN3std10sys_common4once5queue4Once4call17h4d15eaa35c1bfcb7E
foo.lib(foo.foo.5be5606e1f6aa79b-cgu.0.rcgu.o) : error LNK2019: unresolved external symbol __imp_WakeByAddressSingle referenced in function _ZN83_$LT$std..sys_common..once..queue..WaiterQueue$u20$as$u20$core..ops..drop..Drop$GT$4drop17h3b920a1f9cb93053E
C:\a\rust\rust\build\x86_64-pc-windows-msvc\test\run-make\lto-smoke-c\lto-smoke-c\bar.exe : fatal error LNK1120: 2 unresolved externals
make[1]: Leaving directory '/c/a/rust/rust/tests/run-make/lto-smoke-c'
--- stderr -------------------------------
make[1]: *** [Makefile:9: all] Error 2
------------------------------------------

---
    [run-make] tests\run-make\lto-smoke-c

test result: FAILED. 250 passed; 4 failed; 95 ignored; 0 measured; 0 filtered out; finished in 69.44s

Some tests failed in compiletest suite=run-make mode=run-make host=x86_64-pc-windows-msvc target=x86_64-pc-windows-msvc
make: *** [Makefile:73: ci-msvc-ps1] Error 1
  local time: Mon, Feb 26, 2024  1:14:24 AM
##[error]Process completed with exit code 2.
Post job cleanup.
[command]"C:\Program Files\Git\bin\git.exe" version
git version 2.43.0.windows.1
```

</details>

--- a/tests/run-make/tools.mk
+++ b/tests/run-make/tools.mk
@@ -134,9 +134,9 @@ endif
 # Extra flags needed to compile a working executable with the standard library
 ifdef IS_WINDOWS
 ifdef IS_MSVC
-	EXTRACFLAGS := ws2_32.lib userenv.lib advapi32.lib bcrypt.lib ntdll.lib
+	EXTRACFLAGS := ws2_32.lib userenv.lib advapi32.lib bcrypt.lib ntdll.lib synchronization.lib
 else
-	EXTRACFLAGS := -lws2_32 -luserenv -lbcrypt -lntdll
+	EXTRACFLAGS := -lws2_32 -luserenv -lbcrypt -lntdll -lsynchronization
 	EXTRACXXFLAGS := -lstdc++
 	# So this is a bit hacky: we can't use the DLL version of libstdc++ because
 	# it pulls in the DLL version of libgcc, which means that we end up with 2

